cluster/storage/src/main/java/com/linkedin/openhouse/cluster/storage/s3/S3Storage.java

```diff
@@ -34,4 +36,22 @@ public StorageType.Type getType() {
   public StorageClient<?> getClient() {
     return s3StorageClient;
   }
+
+  /**
+   * Deallocates/deletes Table Storage location for S3 storage
+   *
+   * @param location the base location of the table
+   * @param tableCreator the creator of the table
+   * @throws IOException
+   */
+  @Override
+  public void deallocateTableLocation(String location, String tableCreator) throws IOException {
+    try {
+      this.s3StorageClient
+          .getNativeClient()
+          .deleteBucket(DeleteBucketRequest.builder().bucket(location).build());
```

Comment:
currently bucket is defined at cluster level, this code will likely not work for bucket/db/tbl and it'll end up deleting all data.
can we check how S3FileIO implements  SupportsPrefixOperations? and use that code as reference

Reply:
Lets also perform a docker compose test for this in s3